app/scripts/background.js

```diff
@@ -981,45 +981,88 @@ export function setupController(
           connectionStream: portStreamForCookieHandlerPage,
         });
       }
-      connectExternalExtension(remotePort);
+
+      const portStream =
+        overrides?.getPortStream?.(remotePort) || new PortStream(remotePort);
+
+      connectEip1193(portStream, remotePort.sender);
+
+      if (process.env.MULTICHAIN_API && isFirefox) {
+        const mux = setupMultiplex(portStream);
+        mux.ignoreStream(METAMASK_EIP_1193_PROVIDER);
+
+        connectCaipMultichain(
+          mux.createStream(
+            METAMASK_CAIP_MULTICHAIN_PROVIDER,
+            remotePort.sender,
+          ),
+        );
+      }
     }
   };
 
-  // communication with page or other extension
-  connectExternalExtension = (remotePort) => {
+  /**
+   * Connects a externally_connecatable Port to the MetaMask controller.
+   * This method identifies dapp clients and connects them differently from extension clients.
+   *
+   * @param {Port} remotePort - The port provided by a new context.
```

Comment:
Our original code predates the published types for `Port`, but it should be:

```suggestion
   * @param {chrome.runtime.Port} remotePort - The port provided by a new context.
```

OR you could update the `Port` typedef to be an alias of `chrome.runtime.Port` to fix all of them in this file.

Reply:
addressed here https://github.com/MetaMask/metamask-extension/pull/30142/commits/68aacdd7ff2c8d7d37ebdff449ab932e817f0804